core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -105,54 +115,63 @@ static User findUserByMobileNumber(String mobileNumber)
 		return userEntity;
 	}
 
-	private UserDTO handleUserSignupUponBuddyRequest(String password, UserDTO userResource, User originalUserEntity)
+	@Transactional
+	public UserDTO updateUserCreatedOnBuddyRequest(UUID id, String tempPassword, UserDTO userResource)
 	{
-		UpdatedEntities updatedEntities = updateUserWithTempPassword(userResource, originalUserEntity);
-		return saveUserWithDevicePassword(password, updatedEntities);
+		User originalUserEntity = getEntityByID(id);
+		if (!originalUserEntity.isCreatedOnBuddyRequest())
```

Comment:
On this one I think I disagree. If assertion checking is off the password of an existing user may be replaced. I consider this a security issue..